core/lib/dal/src/models/storage_transaction.rs

```diff
@@ -508,6 +506,12 @@ impl StorageApiTransaction {
             .signature
             .and_then(|signature| PackedEthSignature::deserialize_packed(&signature).ok());
 
+        let to = if let Ok(address) = serde_json::from_value(self.execute_contract_address) {
+            Some(address)
+        } else {
+            Some(Address::zero())
+        };
```

Comment:
Why does `to` field have differing fallback logic here and for receipts?

Reply:
I think it keeps the same logic now that the receipt code is changed.